README.md

```diff
@@ -1,12 +1,332 @@
+[![Conda](https://img.shields.io/conda/v/gpytorch/linear_operator.svg)](https://anaconda.org/gpytorch/linear_operator)
+[![PyPI](https://img.shields.io/pypi/v/linear_operator.svg)](https://pypi.org/project/linear_operator)
 
-## Development
-To run unit tests:
+LinearOperator is a PyTorch package for abstracting away the linear algebra routines needed for structured matrices (or operators).
+
+**LinearOperator is in beta.**
+Currently, most of the functionality only supports positive semi-definite and triangular operators.
+
+Package development TODOs:
+ - [x] Support PSD operators
+ - [x] Support triangular operators
+ - [ ] Interface to specify structure (i.e. symmetric, triangular, PSD, etc.)
+ - [ ] Add algebraic routines for symmetric operators
+ - [ ] Add algebraic routines for generic operators
+
+To get started, run either
+```sh
+pip install linear_operator
+# or
+conda install linear_operator -c gpytorch
+```
+or [see below](#installation) for more detailed instructions.
+
+## Why LinearOperator
+Before describing what linear operators are and why they make a useful abstraction, it's easiest to see an example.
+Let's say you wanted to compute a matrix solve:
+
+$$\boldsymbol A^{-1} \boldsymbol b.$$
+
+If you didn't know anything about the matrix $\boldsymbol A$, the simplest (and best) way to accomplish this in code is:
+
+```python
+# A = torch.randn(1000, 1000)
+# b = torch.randn(1000)
+torch.linalg.solve(A, b)  # computes A^{-1} b
+```
+
+While this is easy, the `solve` routine is $\mathcal O(N^3)$, which gets very slow as $N$ grows large.
+
+However, let's imagine that we knew that $\boldsymbol A$ was equal to a low rank matrix plus a diagonal
+(i.e. $\boldsymbol A = \boldsymbol C \boldsymbol C^\top + \boldsymbol D$
+for some skinny matrix $\boldsymbol C$ and some diagonal matrix $\boldsymbol D$.)
+There's now a very efficient $\boldsymbol O(N)$ routine to compute $\boldsymbol A^{-1}$
+called the [Woodbury formula](https://en.wikipedia.org/wiki/Woodbury_matrix_identity).
+**In general**, if we know that $\boldsymbol A$ has structure,
+we want to use efficient linear algebra routines (rather than the general routines)
+that will exploit this structure.
+
+
+### Without LinearOperator
+
+Implementing the efficient solve that exploits the structure in $\boldsymbol A$ would look something like this:
+
+```python
+def low_rank_plus_diagonal_solve(C, d, b):
+    # A = C C^T + diag(d)
+    # A^{-1} b = D^{-1} b - D^{-1} C (I + C^T D^{-1} C)^{-1} C^T D^{-1} b
+    #   where D = diag(d)
+
+    D_inv_b = b / d
+    D_inv_C = C / d.unsqueeze(-1)
+    eye = torch.eye(C.size(-2))
+    return (
+        D_inv_b - D_inv_c @ torch.cholesky_solve(
+            C.mT @ D_inv_b,
+            torch.linalg.cholesky(eye + C.mT @ D_invC, upper=False),
+            upper=False
+        )
+    )
+
+
+# C = torch.randn(1000, 20)
+# d = torch.randn(1000)
+# b = torch.randn(1000)
+low_rank_plus_diagonal_solve(C, d, b)  # computes A^{-1} b efficiently
```

Comment:
Should we say how much more efficiently or is this too much in the weeds?

Reply:
included